src/sentry/migrations/0181_copy_useroptions_to_notificationsettings.py

```diff
@@ -0,0 +1,151 @@
+# Generated by Django 1.11.29 on 2021-03-22 17:36
+
+from django.db import migrations
+from enum import Enum
+
+from sentry.utils.query import RangeQuerySetWrapperWithProgressBar
+
+
+class UserOptionValue:
+    # 'workflow:notifications'
+    all_conversations = "0"
+    participating_only = "1"
+    no_conversations = "2"
+    # 'deploy-emails
+    all_deploys = "2"
+    committed_deploys_only = "3"
+    no_deploys = "4"
+
+
+class ExternalProviders(Enum):
+    GITHUB = 0
+    GITLAB = 1
+    EMAIL = 100
+    SLACK = 110
+
+
+class NotificationScopeType(Enum):
+    USER = 0
+    ORGANIZATION = 10
+    PROJECT = 20
+
+
+class NotificationSettingTypes(Enum):
+    # top level config of on/off
+    # for workflow also includes SUBSCRIBE_ONLY
+    # for deploy also includes COMMITTED_ONLY
+    DEFAULT = 0
+    # send deploy notifications
+    DEPLOY = 10
+    # notifications for issues
+    ISSUE_ALERTS = 20
+    # notifications for changes in assignment, resolution, comments
+    WORKFLOW = 30
+
+
+class NotificationSettingOptionValues(Enum):
+    DEFAULT = 0  # Defer to a setting one level up.
+    NEVER = 10
+    ALWAYS = 20
+    SUBSCRIBE_ONLY = 30  # workflow
+    COMMITTED_ONLY = 40  # deploy
+
+
+def copy_useroption_to_notificationsetting(apps, schema_editor):
+    UserOption = apps.get_model("sentry", "UserOption")
+    User = apps.get_model("sentry", "User")
+    NotificationSetting = apps.get_model("sentry", "NotificationSetting")
+    for user_option in RangeQuerySetWrapperWithProgressBar(UserOption.objects.all()):
+        if user_option.key == "workflow:notifications":
+            # if you have fine tuning for projects, project_id will have a value (rather than None)
+            if user_option.project_id:
+                scope_identifier = user_option.project_id
+                scope_type = NotificationScopeType.PROJECT
+            else:
+                scope_identifier = user_option.user.id
+                scope_type = NotificationScopeType.USER
+            type = NotificationSettingTypes.WORKFLOW
+            if user_option.value == UserOptionValue.all_conversations:
+                value = NotificationSettingOptionValues.ALWAYS
```

Comment:
Same for all your `value` values here, `NotificationSettingOptionValues.ALWAYS.value`. I feel like this migration should probably fail as is, have you tested it locally?

Reply:
Good catch, thank you. I ran it just now, caught some mistakes, and "translated" the values in each table and they look good.